pkg/network/multus.go

```diff
@@ -36,6 +44,47 @@ func changeSafeMultus(prev, next *opv1alpha1.NetworkAddonsConfigSpec) []error {
 	return nil
 }
 
+// cleanUpMultus handles special cleanup of obsoleted multus object
+func cleanUpMultus(ctx context.Context, client k8sclient.Client) []error {
+	errs := []error{}
+
+	//deletes the existing multus object by specific name
+	errs = append(errs, cleanUpMultusOldName(ctx, client)...)
+
+	return errs
+
+}
+
+// cleanUpMultusOldName deletes the Multus ds according to a specific old name and namespace.
+// TODO This function should be dropped once we don't support the version with this specific name & namespace.
+func cleanUpMultusOldName(ctx context.Context, client k8sclient.Client) []error {
+	// Get existing
+	existing := &unstructured.Unstructured{}
+	gvk := schema.GroupVersionKind{Group: "apps", Version: "v1", Kind: "DaemonSet"}
+	existing.SetGroupVersionKind(gvk)
+	namespace := "cluster-network-addons"
+	name := "kube-multus-ds-amd64"
+
+	err := client.Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, existing)
+	if err == nil || !(apierrors.IsNotFound(err) ||
+		strings.Contains(err.Error(), "no matches for kind")) {
```

Comment:
We want to proceed only if `err == nil`. If we failed to `Get` with an unexpected error, we want to return an error and fail the reconciliation. If not found, we can just return, all is good.